### DISCUSSION TITLE
div overflowing parent div in flex -col

### DISCUSSION THREAD

**Discussion by ArgekarStudios** [tailwindlabs/tailwindcss#9618] [Help]
```
 **<div class="flex flex-col w-full h-full bg-red overflow-hidden">
    <div class="flex h-25 bg-blue-100">
      ssds
    </div>
    <div class="flex h-full bg-yellow-200 overflow-y-auto">
      <div class="flex flex-col">
        <div v-for="item in 100" :key="item">
          sd
        </div>
      </div>
    </div>
    <div flex flex-shrink-0 h-2>
      sdsds
    </div>
   </div>
  
  list is overflowing the when it's parent in h-full**
```

**Comment by wongjn** [comment]
If I understand correctly, I believe you would like the `div.bg-yellow-200` element to scroll its content instead of the whole structure being scrolled. You would need to ensure that the parent element for your given DOM structure has a parent with height that it can take `height 100%` from. For example, 100vh for the height of the screen (https://play.tailwindcss.com/WtEBuwmE4Z) but this can be any explicit height.

**Reply by ArgekarStudios** [reply]
but my ui is like [this](https://i.ibb.co/3TnYJrz/Screenshot-from-2022-10-20-21-45-59.png)
https://i.ibb.co/3TnYJrz/Screenshot-from-2022-10-20-21-45-59.png

**Reply by ArgekarStudios** [reply]
I can't use h-screen because it's sub component

**Reply by wongjn** [reply]
As I said, that was just an example. Without any real code, it is hard to suggest any solutions – consider replicating the DOM in a [Tailwind Play](https://play.tailwindcss.com).

**Reply by ArgekarStudios** [reply]
Is there any option without using h-screen

**Reply by wongjn** [reply]
I'm sure there are alternative options, but without any real code, it is hard to suggest any concrete solutions – please consider replicating the DOM in a [Tailwind Play](https://play.tailwindcss.com).

**Reply by ArgekarStudios** [reply]
@wongjn Sloved the issue. Thanks for the heads up

```
<template>
  <div class="flex flex-col w-screen h-screen bg-blue-50 justify-between">
    <div class="flex  flex-shrink-0 w-full h-15 ">
      <TopBar />
    </div>
    <div class="flex w-full h-full flex-row overflow-hidden justify-between">
      <div class="flex flex-col  justify-between mt-10">
        <DevNavigationCard />
        <SignOut />
      </div>
      <div class="flex flex-col w-full ">
        <div flex-none h-10 bg-red />
        <div class="flex w-full flex-col overflow-y-auto">
          <div v-for="items in 100" :key="items">
            {{ items }}
          </div>
        </div>
        <div flex-none h-10 bg-blue />
      </div>

      <div w-1 bg-black />
    </div>
    <div h-3 bg-green />
  </div>
</template>

```
